morango/utils/register_models.py

```diff
@@ -50,29 +51,34 @@ def __new__(mcls, name, bases, namespace):
             return cls
 
     class DatabaseMaxCounter(six.with_metaclass(MorangoIndexedModelMeta, AbstractDatabaseMaxCounter)):
-        """
-        `DatabaseMaxCounter` is used to keep track of what data this database already has across all
-        instances for a particular filter.
-        """
 
         class Meta:
             app_label = label
 
-    class DataTransferBuffer(six.with_metaclass(MorangoIndexedModelMeta, AbstractStoreModel)):
-        """
-        ``DataTransferBuffer`` is where records from the internal store are kept temporarily,
-        until they are sent to another morango instance.
-        """
+    class DataTransferBuffer(six.with_metaclass(MorangoIndexedModelMeta, AbstractStore)):
 
-        transfer_session_id = UUIDField()
+        incoming_buffer = models.BooleanField()
+        transfer_session = models.ForeignKey(TransferSession, related_name='{}_buffer'.format(profile))
 
         class Meta:
             app_label = label
 
-    class StoreModel(six.with_metaclass(MorangoIndexedModelMeta, AbstractStoreModel)):
-        """
-        ``StoreModel`` is where serialized data is persisted, along with metadata about counters and history.
-        """
+    class StoreModel(six.with_metaclass(MorangoIndexedModelMeta, AbstractStore)):
+
+        class Meta:
+            app_label = label
+
+    class RecordMaxCounter(AbstractRecordMaxCounter):
+
+        store_model = models.ForeignKey(StoreModel)
+
+        class Meta:
+            app_label = label
+
+    class RecordMaxCounterBuffer(RecordMaxCounter):
```

Comment:
I think for these we decided maybe not, because the PK is auto-generated and so we can't set it in the single-SQL-query transfer. But I'm not sure what the alternative is (I guess storing both the transfer ID and the UUID) -- let's chat about this piece.